### DISCUSSION TITLE
How to hide one test's failure details in Failures section from within a test

### DISCUSSION THREAD

**Discussion by ari-s** [pytest-dev/pytest#11870] [Q&A]
suppose I have a test like this:
```python
import random
def test_example():
    assert 0 == random.randrange(1, 9)
```

running this test with `pytest /tmp/test.py` will produce:

```
==================================================== FAILURES ====================================================
__________________________________________________ test_example __________________________________________________

    def test_example():
>       assert 0 == random.randrange(1, 9)
E       assert 0 == 2
E        +  where 2 = <bound method Random.randrange of <random.Random object at 0x1798b70>>(1, 9)
E        +    where <bound method Random.randrange of <random.Random object at 0x1798b70>> = random.randrange

/tmp/test2.py:3: AssertionError
============================================ short test summary info =============================================
FAILED /tmp/test.py::test_example - assert 0 == 2

```

in the commandline I can add `--tb=no` to hide the details of the failure. 
**Is it possible to trigger this output modification from within the code?**
A constant decision to hide would be good, a callable would be perfect.
I've tried `__tracebackhide__`, but it doesn't have the desired effect. Even though from the names it sounds like it would do the same as `--tb=no`

Background: I have several thousand test cases through parametrisation. Many of them are equivalent, but it's somewhat hard to weed out the equivalent cases but retain the full coverage.

**Comment by Colelyman** [comment]
You might be able to find something useful in the [Skipping tests section](https://docs.pytest.org/en/stable/how-to/skipping.html) (in case you haven't read that yet).

To follow your example, you could mark this test as expected to fail, like this:

```python
import random
import pytest

@pytest.mark.xfail
def test_example():
    assert 0 == random.randrange(1, 9)
```

You also mentioned that a callable would be preferred, you could use `pytest.xfail('Message')` like this:

```python
import random
import pytest

@pytest.mark.xfail
def test_example():
    try:
        fail = 1 / 0
    except Exception:
        pytest.xfail('Expected to fail')
```

Note that all `xfail`s will still show up in the summary as `xfailed`, but the overall suite will still pass.
